Framework/Core/include/Framework/HistogramRegistry.h

```diff
@@ -142,13 +179,26 @@ class HistogramRegistry
   mutable uint32_t lookup = 0;
 
  private:
+  template <typename T>
```

Comment:
Why making a template method, rather than a templated helper? Something like:

```
registry.insert(hs<TH1>());
```

This way you move the complexity on the helper rather than on the registry, which, IMHO it's more manageable / customizable.

Reply:
OK, then we are back to making HistogramSpec templated? IMHO HistogramSpec needs to be templated to create different histogram types in a nicer way.